src/Agent.Listener/Configuration/WindowsSecurityManager.cs

```diff
@@ -0,0 +1,477 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.AccessControl;
+using System.Security.Cryptography;
+using System.Security.Principal;
+using System.Text;
+
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator (Default = typeof(WindowsSecurityManager))]
+    public interface IWindowsSecurityManager : IAgentService
+    {
+        string GetUniqueBuildGroupName();
+
+        bool LocalGroupExists(string groupName);
+
+        void CreateLocalGroup(string groupName);
+
+        void AddMemberToLocalGroup(string accountName, string groupName);
+
+        void GrantFullControlToGroup(string path, string groupName);
+
+        bool GrantUserLogonAsServicePrivilage(string accountName);
+
+        bool IsValidCredential(string logonAccount, string logonPassword);
+
+        NTAccount GetDefaultServiceAccount();
+
+        void SetPermissionForAccount(string path, string accountName);
+    }
+
+    public class WindowsSecurityManager : AgentService, IWindowsSecurityManager
+    {
+        private const string AgentServiceLocalGroupPrefix = "VSTS_AgentService_G";
+
+        public string GetUniqueBuildGroupName()
+        {
+            return AgentServiceLocalGroupPrefix + GetHashCodeForAgent().Substring(0, 5);
+        }
+
```

Comment:
Focus on, If an error happens, can the user debug himself by looking at log and enough information give in console. Thats what we are trying to achieve right? Like your other comment why hide win32api, lets give it to user so that he could fix it was very useful. I guess your review is already good, take a second look. I have updated the PR.